Source/WebKit/Shared/WebCoreArgumentCoders.cpp

```diff
@@ -3220,4 +3223,55 @@ void ArgumentCoder<PixelBuffer>::encode<Encoder>(Encoder&, const PixelBuffer&);
 template
 void ArgumentCoder<PixelBuffer>::encode<StreamConnectionEncoder>(StreamConnectionEncoder&, const PixelBuffer&);
 
+void ArgumentCoder<Ref<WebCore::Report>>::encode(Encoder& encoder, const Ref<WebCore::Report>& report)
+{
+    report->encode(encoder);
+}
+
+std::optional<Ref<WebCore::Report>> ArgumentCoder<Ref<WebCore::Report>>::decode(Decoder& decoder)
+{
+    return WebCore::Report::decode(decoder);
+}
+
+void ArgumentCoder<RefPtr<WebCore::ReportBody>>::encode(Encoder& encoder, const RefPtr<WebCore::ReportBody>& reportBody)
+{
+    bool hasReportBody = !!reportBody;
+    encoder << hasReportBody;
+    if (!hasReportBody)
+        return;
+
+    encoder << reportBody->reportBodyType();
+
+    switch (reportBody->reportBodyType()) {
+    case ReportBodyType::CSPViolation:
+        downcast<CSPViolationReportBody>(reportBody.get())->encode(encoder);
+        return;
+    }
+
+    RELEASE_ASSERT_NOT_REACHED();
+}
+
+std::optional<RefPtr<WebCore::ReportBody>> ArgumentCoder<RefPtr<WebCore::ReportBody>>::decode(Decoder& decoder)
+{
+    bool hasReportBody = false;
+    if (!decoder.decode(hasReportBody))
+        return std::nullopt;
+
+    RefPtr<WebCore::ReportBody> result;
+    if (!hasReportBody)
+        return { result };
+
+    std::optional<ReportBodyType> reportBodyType;
+    decoder >> reportBodyType;
+    if (!reportBodyType)
+        return std::nullopt;
+
+    switch (*reportBodyType) {
+    case ReportBodyType::CSPViolation:
+        return CSPViolationReportBody::decode(decoder);
+    }
+
+    RELEASE_ASSERT_NOT_REACHED();
```

Comment:
I don't think we want to release assert here.
If a rogue web content process sends a bad IPC, we shouldn't crash the entire Safari.

Reply:
Good point -- I'll treat it as a serialization failure and return std::nullopt and use a debug assert.